src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/NftMarketplaceCustomizationForm.tsx

```diff
@@ -20,13 +23,33 @@ export function NftMarketplaceCustomizationForm({
     projectMetadata: { name: projectName },
   } = useAppSelector(state => state.editingV2Project)
 
-  const initialValues = {
-    collectionName: name,
-    collectionDescription: description,
-    collectionSymbol: symbol,
-  }
+  const initialValues = useMemo(
+    () => ({
+      collectionName: name,
+      collectionDescription: description,
+      collectionSymbol: symbol,
+    }),
+    [name, description, symbol],
+  )
+
+  const handleFormChange = useCallback(() => {
+    const hasUpdated =
+      initialValues.collectionName !== form.getFieldValue('collectionName') ||
+      initialValues.collectionDescription !==
+        form.getFieldValue('collectionDescription') ||
+      initialValues.collectionSymbol !== form.getFieldValue('collectionSymbol')
+    onFormUpdated(hasUpdated)
+  }, [form, initialValues, onFormUpdated])
+
+  useEffect(() => handleFormChange(), [handleFormChange])
+
   return (
-    <Form layout="vertical" form={form} initialValues={initialValues}>
+    <Form
+      layout="vertical"
+      form={form}
+      initialValues={initialValues}
+      onValuesChange={handleFormChange}
```

Comment:
I'm about to be annoying here, and feel free to not enact this change (especially if this is working), but from a clean perspective, I think we should lift this logic (checking if form has updated) to its parent as it is (imo) not really a concern of the form itself. 

The parent component (`NftDrawer`) is actually the one that cares about the state changing.

We can make an issue for this (or solve here, depending on how you feel) but I think what would be better is that `NftMarketplaceCustomizationForm` mirrors the result of `Form.onValuesChange` and returns that.

`Form.onValuesChange` returns a subset of what values have actually changed.

This is actually a non-trivial change from what you have so I actually recommend not making this change here - lets a add a low priority issue instead - I can make it, but i'll wait to see what you think as you might disagree ;-)

Reply:
Also happy to jump on a call to discuss this if it helps as I feel like I have kind of just jumbled a few incoherent sentences together lol

Reply:
I like this idea @wraeth-eth, it'll be a good cleanup task. It will also apply to all the other drawers, correct? The new issue could tackle them all I guess